packages/edge-config/src/index.common.test.ts

```diff
@@ -27,168 +17,164 @@ describe('test conditions', () => {
-          status: 304,
-          headers: { ETag: 'a' },
-        });
+        expect(fetchMock).toHaveBeenCalledTimes(1);
+        expect(fetchMock).toHaveBeenCalledWith(
+          `${modifiedBaseUrl}/item/foo?version=1`,
+          {
+            headers: new Headers({ Authorization: 'Bearer token-2' }),
+            cache: 'no-store',
+          },
+        );
+      });
+    });
+  });
 
-        // second call should reuse response
+  describe('has(key)', () => {
+    describe('when item exists', () => {
+      it('should return true', async () => {
+        fetchMock.mockResponse('');
 
-        await expect(edgeConfig.get('foo')).resolves.toEqual('bar');
+        await expect(edgeConfig.has('foo')).resolves.toEqual(true);
 
-        expect(fetchMock).toHaveBeenCalledTimes(2);
+        expect(fetchMock).toHaveBeenCalledTimes(1);
         expect(fetchMock).toHaveBeenCalledWith(
           `${modifiedBaseUrl}/item/foo?version=1`,
           {
+            method: 'HEAD',
             headers: new Headers({ Authorization: 'Bearer token-2' }),
             cache: 'no-store',
           },
         );
+      });
+    });
+  });
+
+  describe('digest()', () => {
+    describe('when the request succeeds', () => {
+      it('should return the digest', async () => {
+        fetchMock.mockResponse(JSON.stringify('awe1'));
+
+        await expect(edgeConfig.digest()).resolves.toEqual('awe1');
+
+        expect(fetchMock).toHaveBeenCalledTimes(1);
         expect(fetchMock).toHaveBeenCalledWith(
-          `${modifiedBaseUrl}/item/foo?version=1`,
+          `${modifiedBaseUrl}/digest?version=1`,
           {
-            headers: new Headers({
-              Authorization: 'Bearer token-2',
-              'if-none-match': 'a',
-            }),
+            headers: new Headers({ Authorization: 'Bearer token-2' }),
             cache: 'no-store',
           },
         );
       });
     });
   });
 });
+
+describe('etags and if-none-match', () => {
+  const modifiedConnectionString =
+    'https://edge-config.vercel.com/ecfg-2?token=token-2';
+  const modifiedBaseUrl = 'https://edge-config.vercel.com/ecfg-2';
+  let edgeConfig: EdgeConfigClient;
+
+  beforeEach(() => {
+    fetchMock.resetMocks();
+    cache.clear();
+    edgeConfig = pkg.createClient(modifiedConnectionString);
+  });
+
+  describe('when reading the same item twice', () => {
+    it('should reuse the response', async () => {
+      fetchMock.mockResponseOnce(JSON.stringify('bar'), {
+        headers: { ETag: 'a' },
+      });
+
+      await expect(edgeConfig.get('foo')).resolves.toEqual('bar');
+
+      // the server would not actually send a response body the second time
+      // as the etag matches
+      fetchMock.mockResponseOnce('', {
+        status: 304,
+        headers: { ETag: 'a' },
+      });
+
+      // second call should reuse response
+
+      await expect(edgeConfig.get('foo')).resolves.toEqual('bar');
+
+      expect(fetchMock).toHaveBeenCalledTimes(2);
```

Comment:
Should we assert that readFile is not called as we do in the node tests?

Reply:
If it was called in edge it would throw, so I think we're good.

We can follow up in another PR if we want to clean up more 👍